src/main.cpp

```diff
@@ -2026,8 +2026,18 @@ bool CScriptCheck::operator()() {
 int GetSpendHeight(const CCoinsViewCache& inputs)
 {
     LOCK(cs_main);
-    CBlockIndex* pindexPrev = mapBlockIndex.find(inputs.GetBestBlock())->second;
-    return pindexPrev->nHeight + 1;
+    BlockMap::iterator i = mapBlockIndex.find(inputs.GetBestBlock());
+    if (i != mapBlockIndex.end())
+      {
+      CBlockIndex* pindexPrev = i->second;
+      if (pindexPrev)
+        return pindexPrev->nHeight + 1;
+      else
+        {
+          throw runtime_error("GetSpendHeight(): mapBlockIndex contains null block");
```

Comment:
an exception in the P2P handling code path will be caught outside the message processing with the effect that a specific message will be ignored, but the process will continue.  an exception in other threads will often go all the way out.